Upgarding to JBehave 4.0.5, JUnit 4.12, Mockito 1.10.19. Adding support for @BeforeStory/@AfterStory and @BeforeScenario/@AfterScenario.

valfirst opened the pull request:
Not sure if you are interested in these changes, but I was needed them to handle properly failures in @Before / @After methods. Also I've made a breaking changes: upgraded to JBehave 4.0.3.
- Upgarding to 
  - JBehave 4.0.5
  - JUnit 4.12.
  - Mockito 1.10.19.
- Adding support for @BeforeStory/@AfterStory and @BeforeScenario/@AfterScenario
- Fix for issue #80 "JUnitScenarioReporter throws NPE when story file name contains parentheses"
- Filtered out stories and scenarios shouldn't be included in JUnit description tree
- Refactor and optimize creation of configuredEmbedder and candidateSteps
- Search for "storyPaths" method in superclasses > 
  "Previously "storyPaths" method was searched in all methods of the test
  class (with any visibility modifier) and in public methods of test class
  superclasses. Now all methods of superclasses are included in the search.
  It may be useful for cases when there is some common class with protected
  "storyPaths" method and there is a set of sub-classes adding custom logic,
  but not overriding "storyPaths" method."
- Use ExamplePerformableScenario to generate descritpions for examples >     ExamplePerformableScenario suits better for generation of descriptions for examples than ExamplesTable from scenario, since JBehave may apply additional operations which are reflected only in ExamplePerfoemableScenario

valfirst committed:
Upgarding to JBehave 4.0.3, JUnit 4.12, Mockito 1.10.19. Adding support for @BeforeStory/@AfterStory and @BeforeScenario/@AfterScenario.

britter commented:
Hello @valfirst,

looks like you have introduced new code which is not sufficiently covered by unit tests. Can you provide some more test coverage?

Thank you!

valfirst commented:
Hello @britter,

Sure, I'll add more tests soon, it looks like my code coverage plugin works in a little bit different way

Thanks

valfirst commented:
@britter, could you, please, review the changes?

valfirst committed:
Fix for issue #80 "JUnitScenarioReporter throws NPE when story file name contains parentheses"

valfirst committed:
Filtered out stories and scenarios shouldn't be included in JUnit description tree

valfirst committed:
Upgrade to Jbehave 4.0.5

valfirst committed:
Refactor and optimize creation of configuredEmbedder and candidateSteps

valfirst commented:
Does anybody review changes and merge pull-requests? Or is this project dead?

valfirst committed:
Search for "storyPaths" method in superclasses

Previously "storyPaths" method was searched in all methods of the test
class (with any visibility modifier) and in public methods of test class
superclasses. Now all methods of superclasses are included in the search.
It may be useful for cases when there is some common class with protected
"storyPaths" method and there is a set of sub-classes adding custom logic,
but not overriding "storyPaths" method.

valfirst committed:
Use ExamplePerformableScenario to generate descritpions for examples

ExamplePerformableScenario suits better for generation of descriptions
for examples than ExamplesTable from scenario, since JBehave may apply
additional operations which are reflected only in ExamplePerfoemableScenario

valfirst commented:
@britter, is this project dead?

britter commented:
@valfirst as far as I know there is currently nobody actively maintaining this project at codecentric.

valfirst commented:
@britter, is there any way to transfer ownership or is it better to proceed with fork development?

valfirst commented:
> @britter, is there any way to transfer ownership or is it better to proceed with fork development?

@britter?

britter commented:
@valfirst I'm currently checking with the company whether how to proceed.

valfirst commented:
@britter thank you! looking forward for updates from you

AndreasEK commented:
Thanks for your efforts, just merged you pull request back into master :)

valfirst commented:
@AndreasEK, cool, thank you! Are you going to maintain this project? I would like to take participation in development of this project by providing more updates and fixes
